vividus-extension-selenium/src/test/java/org/vividus/selenium/screenshot/ScreenshotUtilsTests.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.selenium.screenshot;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.WebDriver;
+import org.vividus.util.ResourceUtils;
+
+@ExtendWith(MockitoExtension.class)
+class ScreenshotUtilsTests
+{
+    @Mock(extraInterfaces = TakesScreenshot.class)
+    private WebDriver webDriver;
+
+    @Test
+    void shouldTakeScreenshotAsByteArray()
+    {
+        byte[] bytes = mockScreenshotTaking();
+
+        assertArrayEquals(bytes, ScreenshotUtils.takeScreenshotAsByteArray(webDriver));
+    }
+
+    @Test
+    void shouldTakeViewportScreenshot() throws IOException
+    {
+        mockScreenshotTaking();
+
+        BufferedImage image = ScreenshotUtils.takeViewportScreenshot(webDriver);
+
+        assertEquals(400, image.getWidth());
+        assertEquals(600, image.getHeight());
+    }
+
+    private byte[] mockScreenshotTaking()
+    {
+        byte[] bytes = ResourceUtils.loadResourceAsByteArray(getClass(), "image.png");
```

Comment:
Do we really need another IO during unit tests? :) You are mocking WD method with the data loaded from the file, when all you need to verify its method call